calico_containers/pycalico/block.py

```diff
@@ -129,12 +142,20 @@ def from_etcd_result(cls, etcd_result):
         json_dict = json.loads(etcd_result.value)
         cidr_prefix = IPNetwork(json_dict[AllocationBlock.CIDR])
 
-        # Parse out the host.  For now, it's in the form host:<host id>
+        # Parse out the host.  For now, it's in the form host:<host id>.  An
+        # empty host ID is converted to None to indicate the block has no
+        # specific host affinity.
         affinity = json_dict[AllocationBlock.AFFINITY]
         assert affinity[:5] == "host:"
-        host_affinity = affinity[5:]
+        host_affinity = affinity[5:] or None
 
-        block = cls(cidr_prefix, host_affinity)
+        # Parse out the strict_host_affinity flag.  If this does not exist
+        # assume False.
+        strict_host_affinity = json_dict.get(
```

Comment:
Should this field actually be called `strict_host_affinity` or should it be called `strict_affinity`?

Even though our IPAM code only supports host affinity currently, I believe the data model has been written to support various other types (e.g rack) affinity as well.